### ISSUE TITLE
TwoFactorAuthenticationCodeNotificationEvent not triggering correctly

### ISSUE THREAD

**Issue by dpapada** [brockallen/BrockAllen.MembershipReboot#488]
Using verison 8.1.0, I am currently attempting to develop a form of two-factor authentication with SMS, using a custom event handler. Currently the way i have it setup is my custom user service login detects if twoFactor is enabled and then uses a partial login flow to direct the user to the two-factor, which is all working successfully. Within that flow I am using the following code before redirecting:  userAccountService.SendTwoFactorAuthenticationCode(account.ID);

The expectation is that I thought this would trigger the TwoFactorAuthenticationCodeNotificationEvent in hopes that after triggered that would use my SMS functions to send the 6 digit code.

However, it doesn't seem to make it past this conditonal statement in the UserAccountService class:

**Comment by dpapada** [comment]
My apologies, I got to quick on the the keyboard.

 if (CanResendMobileCode(account) ||
                account.CurrentTwoFactorAuthStatus != TwoFactorAuthMode.Mobile)

Is there some way to re-send the the MobileCode other than I am doing it?

**Comment by brockallen** [comment]
The reason it's that way is because the 6-digit mobile code is stored in a non-revsersible way in the DB, so we can't re-send the original code. We can only send a new code. This is why we throttle it with the CanResend API.

**Comment by dpapada** [comment]
Hi Brock,

Thanks for that information. I understand that completely,
 I did some more digging and realized I wasn't meeting the minimum time requirements, in some cases. I was debugging my process flow, and after I enabled two-factor mode for a particular test user, I immediately went to my login screen and attempted a login in which I would fail on the condition to wait for i think it was 1 or 2 minutes after first enabling. Thus the reason for it not triggering the event...I am now getting the TwoFactorAuthenticationCodeNotificationEvent triggered as expected.

I apologize, I should have commented on this earlier and closed it.
